sbi/inference/snpe/base_snpe.py

```diff
@@ -100,9 +109,11 @@ def __init__(
             self.obs_std = torch.ones(self._true_observation.shape)
 
         # new embedding_net contains z-scoring
-        if not isinstance(self._neural_posterior, MDNPosterior):
+        if not isinstance(self._neural_posterior.neural_net, MultivariateGaussianMDN):
+            print("surprise")
             embedding = nn.Sequential(
```

Comment:
Is there a reason for setting the embedding net post-hoc rather than passing it when the Posterior is constructed?